app/src/lib/lookups.js

```diff
@@ -90,27 +89,130 @@ async function check_forms(lookup_token) {
 
 	// Reattach the sense if present
 	lookup_token.lookup_terms = sense ? [...unique_stems].map(stem => `${stem}-${sense}`) : [...unique_stems]
+
+	/**
+	 * 
+	 * @param {FormResult} form_result 
+	 * @returns {LookupResult}
+	 */
+	function transform_result(form_result) {
+		return create_lookup_result(form_result, { form: form_result.form })
+	}
 }
 
 /**
  * @param {Token} lookup_token
  */
 async function check_ontology(lookup_token) {
-	const results = await Promise.all(lookup_token.lookup_terms.map(check_word_in_ontology))
-	lookup_token.lookup_results = results.flat()
+	const results = (await Promise.all(lookup_token.lookup_terms.map(check_word_in_ontology))).flat()
+
+	const found_results = results.map(transform_result)
+	const not_found_results = lookup_token.lookup_results.filter(lookup => !results.some(result => lookups_match(lookup, result)))
+	lookup_token.lookup_results = found_results.concat(not_found_results)
 
 	/**
 	 * 
 	 * @param {string} lookup 
 	 * @returns {Promise<OntologyResult[]>}
 	 */
 	async function check_word_in_ontology(lookup) {
-		const response = await fetch(`/ontology-lookup?word=${lookup}`)
+		const response = await fetch(`/lookup/ontology?word=${lookup}`)
 
-		/** @type {LookupResult<OntologyResult>} */
+		/** @type {LookupResponse<OntologyResult>} */
 		const results = await response.json()
 		let matches = results.matches
 
 		return matches
 	}
+
+	/**
+	 * 
+	 * @param {OntologyResult} ontology_result 
```

Comment:
```suggestion
	 * @param {OntologyResult} concept 
```

I'm not certain if this is better or not but it would simplify your `create_lookup_result` call later...

Reply:
I like it as `ontology_result` here because it's fresh from the api response. Once it gets attached to the lookup result then it can be named `concept`.